### PR TITLE
Ire of the Betrayed - The Hiskyn Revanchist Ship

### PR DESCRIPTION
**Pull request by RustingWithYou**
Adds a [Hiskyn's Revanchists](https://wiki.aurorastation.org/index.php?title=Unathi_Piracy#Hiskyn's_Revanchists) pirate ship, complete with ghostroles. Also moves the Izharshan and Tarwa pirate ships from the away sites to ships folder.

### CODE REVIEW

**Comment by RustingWithYou** [comment]
!review

**Review by NonQueueingMatt — APPROVED** [review]

**Comment by DreamySkrell** [comment]
image
image
image

shuttle is not compliant with dock/shuttle standards

also the two air tanks on the shuttle don't seem to be connected to anything

**Comment by DreamySkrell** [comment]
image

the center hallway should have some firedoors dividing it into parts

**Comment by DreamySkrell** [comment]
image
the landmarks should have types and names that are more informative

instead of:
```
/obj/effect/shuttle_landmark/hiskyn/dock4
	name = "Aft Port Dock"
	landmark_tag = "nav_hiskyn_dock4"
```
should be:
```
/obj/effect/shuttle_landmark/hiskyn/dock/aft_port
	name = "Aft Port Dock"
	landmark_tag = "nav_hiskyn_dock_aft_port"
```

**Comment by DreamySkrell** [comment]
image
the engine nacelles are sad and empty
put some trash, empty crates, materials, supplies, a tool on a table, etc

**Comment by DreamySkrell** [comment]
image
image
thrusters should be separated by a wall from the interior

**Comment by DreamySkrell** [comment]
image
I could be blind and tired but I see only 2 spawn points for the crew, while the limit is 4

**Comment by DreamySkrell** [comment]
image
In general though, the ship lacks basically any kind of decorations. Like I get the intention is that it's like a patchwork ship fused together from multiple separate ships or shuttles, but it just looks kinda bland and uninteresting honestly. Like the core of the ship is a super long 3-tile-wide hallway. 

Should fix the other things, but I'm not going to stop this from being merged cause of lack of decorations.

**Comment by DreamySkrell** [comment]
image
The ship does not seem to be centered on the map. Not sure if this is just mapdiffbot or if it's actually like that.

**Comment by DreamySkrell** [comment]
image
optional, but would not be hard to add a dock here, by just moving atmos to one of the thruster rooms

**Comment by DreamySkrell** [comment]
image
there is no way to fill this big CO2 tank

there should also be some designated storage for empty/filled canisters, could use one of the thruster rooms for that

**Comment by DreamySkrell** [comment]
image
captains office room could use some decorations, chair(s) for the vistitor(s), a pen, a locker or cabinet, a plant maybe (iirc there's a moghes potted plant)

**Comment by DreamySkrell** [comment]
image
not sure if there's cooking tools there, like bowls or rolling thingies or kitchen knife

**Review by DreamySkrell — CHANGES_REQUESTED** [review]
as above

File: maps/away/ships/unathi_pirate/hiskyn/unathi_pirate_hiskyn_ghostroles.dm [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/datum/ghostspawner/human/hiskyn
+	short_name = "hiskyn"
+	name = "Hiskyn's Revanchist Crewman"
+	desc = "Serve aboard a ship of the pirate fleet Hiskyn's Revanchists - former Kazhkz-Han'san Unathi who refused to join the Empire of Dominia. Obey your captain, defend your ship and crew, and keep an eye out for opportunities to shed some Kazhkz-Han'san blood! NOT AN ANTAGONIST! Do not act as such."
```
Comment (DreamySkrell) [inline maps/away/ships/unathi_pirate/hiskyn/unathi_pirate_hiskyn_ghostroles.dm:4]: all ghost spawners should have separate desc and welcome_message
see goon base for an example

File: maps/away/ships/unathi_pirate/hiskyn/unathi_pirate_hiskyn.dm [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	logging_home_tag = "nav_dock_hiskyn"
+	defer_initialisation = TRUE
+
+/obj/effect/shuttle_landmark/hiskyn_shuttle/dock
+	name = "Primary Docking Port"
+	landmark_tag = "nav_dock_hiskyn"
+	base_area = /area/space
+	base_turf = /turf/space
+	docking_controller = "hiskyn_shuttle_dock"
+	movable_flags = MOVABLE_FLAG_EFFECTMOVE
```
Comment (DreamySkrell) [inline maps/away/ships/unathi_pirate/hiskyn/unathi_pirate_hiskyn.dm:129]: why?

File: maps/away/ships/unathi_pirate/hiskyn/unathi_pirate_hiskyn_areas.dm [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+/area/hiskyn_ship/bridge
+	name = "Revanchist Ship - Bridge"
+	icon_state = "bridge"
+
+/area/hiskyn_ship/hallf
+	name = "Revanchist Ship - Fore Hall"
+	icon_state = "hallF"
+
+/area/hiskyn_ship/hallC
```
Comment (DreamySkrell) [inline maps/away/ships/unathi_pirate/hiskyn/unathi_pirate_hiskyn_areas.dm:18]: all types should be all lowercase
hall_c instead of hallC

**Review by DreamySkrell — CHANGES_REQUESTED** [review]
as above

**Comment by RustingWithYou** [comment]
> image I could be blind and tired but I see only 2 spawn points for the crew, while the max count in the code is 4

there are two more in the beds

**Comment by RustingWithYou** [comment]
> image image image
> 
> shuttle is not compliant with dock/shuttle standards
> 
> also the two air tanks on the shuttle don't seem to be connected to anything

this is a restricted dock for the hiskyn shuttle, does being able to fit the max shuttle size really matter here when this is the only shuttle that will ever dock with it?

**Comment by DreamySkrell** [comment]
image
same thing about about the ship thrusters applies to the shuttle as well

image
also should be some kind of floor under ship thrusters

**Comment by DreamySkrell** [comment]
image
why is the scrubber pipe net connected like this? It would pump waste air to a (full) air tank, should be empty canister instead like on many other shuttles

**Comment by DreamySkrell** [comment]
> the air tank thing is true though, autopiper must not connect to pipe adapters

Also for the future, the autopiper does not connect to the universal pipe adapters, cause there isn't really a way to know when/if it should add the connection.

Like in this case, there is no way to know if you want red pipes connected to the adapter above, or that the blue pipes should be connected to the adapter below. It could always connect both blue and red to every adapter, but that would be even more wrong than not connecting at all.

**Comment by RustingWithYou** [comment]
> > > image image image
> > > shuttle is not compliant with dock/shuttle standards
> > > also the two air tanks on the shuttle don't seem to be connected to anything
> > 
> > 
> > this is a restricted dock for the hiskyn shuttle, does being able to fit the max shuttle size really matter here when this is the only shuttle that will ever dock with it?
> > the air tank thing is true though, autopiper must not connect to pipe adapters
> 
> image
> 
> the problem isn't with the dock but with the shuttle, it does not fit inside the shuttle size rectangle

My bad, addressed.

**Review by DreamySkrell — APPROVED** [review]

**Pull request merged**